src/main/kotlin/com/sourcegraph/cody/agent/intellij_extensions/Document.kt

```diff
@@ -13,6 +13,14 @@ fun Document.codyPosition(offset: Int): Position {
 }
 
 fun Document.codyRange(startOffset: Int, endOffset: Int): Range {
+  if (startOffset < 0 ||
+      startOffset > this.textLength ||
+      endOffset > this.textLength ||
+      startOffset > endOffset) {
+    throw IllegalArgumentException(
```

Comment:
Initially I wanted to bound the values here. But maybe we should not do that? We should be aware of what arguments we pass here.